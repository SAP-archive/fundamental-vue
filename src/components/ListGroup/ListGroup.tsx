import Vue from 'vue';
import { Component, DefaultSlot } from '@/core';

@Component('ListGroup')
@DefaultSlot('List of list group items.')
export class ListGroup extends Vue {
  public render() {
    return (<ul class='fd-list-group'>{this.$slots.default}</ul>);
  }
}
