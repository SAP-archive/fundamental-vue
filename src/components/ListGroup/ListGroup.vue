<template>
  <ul class="fd-list-group">
    <slot />
  </ul>
</template>

<script>
export default {
  name: "FdListGroup"
};
</script>
