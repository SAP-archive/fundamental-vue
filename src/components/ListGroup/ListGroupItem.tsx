import Vue from 'vue';
import { Slot, Component, DefaultSlot } from '@/core';

@Component('ListGroupItem')
@DefaultSlot('Content displayed by the item. Usually text.')
@Slot('action', 'Custom actions (displayed on the right side, usually a button)')
export class ListGroupItem extends Vue {
  public render() {
    const action = this.$slots.action;
    return (
      <li class='fd-list-group__item'>
        {this.$slots.default}
        {!!action && <span class='fd-list-group__action'>{action}</span>}
      </li>
    );
  }
}
