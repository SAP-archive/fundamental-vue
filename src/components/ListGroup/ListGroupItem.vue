<template>
  <li class="fd-list-group__item">
    <slot />
    <span v-if="hasAction" class="fd-list-group__action">
      <slot name="action" />
    </span>
  </li>
</template>

<script>
export default {
  name: "FdListGroupItem",
  computed: {
    hasAction() {
      return this.$slots.action != null;
    }
  }
};
</script>
