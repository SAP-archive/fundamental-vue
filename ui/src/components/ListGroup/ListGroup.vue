<template>
  <ul class="fd-list-group">
    <slot />
  </ul>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "FdListGroup"
});
</script>
