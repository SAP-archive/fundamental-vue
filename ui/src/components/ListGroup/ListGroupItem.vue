<template>
  <li class="fd-list-group__item">
    <slot />
    <span v-if="hasAction" class="fd-list-group__action">
      <slot name="action" />
    </span>
  </li>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "FdListGroupItem",
  computed: {
    hasAction(): boolean {
      return this.$slots.action != null;
    }
  }
});
</script>
