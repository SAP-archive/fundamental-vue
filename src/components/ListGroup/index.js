import ListGroup from "./ListGroup.vue";
import ListGroupItem from "./ListGroupItem.vue";
import { pluginify } from "./../../util";
export default pluginify(ListGroup, ListGroupItem);
export { ListGroup, ListGroupItem };
