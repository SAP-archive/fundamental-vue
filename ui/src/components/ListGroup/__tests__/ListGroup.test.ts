import { mount } from "@vue/test-utils";
import ListGroup from "../ListGroup.vue";
import ListGroupItem from "../ListGroupItem.vue";
import { VNode, CreateElement } from "vue";

describe("ListGroup", () => {
  const createListElement = (h: CreateElement, index: number) => {
    const actionElement = h(
      "div",
      {
        slot: "action"
      },
      `Action ${index}`
    );
    return h(ListGroupItem, {}, [actionElement, `List Item ${index}`]);
  };

  const wrapper = mount(ListGroup, {
    slots: {
      default: {
        render(h: CreateElement): VNode {
          const listElement1 = createListElement(h, 1);
          const listElement2 = createListElement(h, 2);
          return h("div", {}, [listElement1, listElement2]);
        }
      }
    }
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has correct number of list items", () => {
    expect(wrapper.findAll("li").length).toBe(2);
  });
});
