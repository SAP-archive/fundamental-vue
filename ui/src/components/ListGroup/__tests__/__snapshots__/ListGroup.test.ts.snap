// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup renders correctly 1`] = `
<ul
  class="fd-list-group"
>
  <div>
    <li
      class="fd-list-group__item"
    >
      List Item 1 
      <span
        class="fd-list-group__action"
      >
        <div>
          Action 1
        </div>
      </span>
    </li>
    <li
      class="fd-list-group__item"
    >
      List Item 2 
      <span
        class="fd-list-group__action"
      >
        <div>
          Action 2
        </div>
      </span>
    </li>
  </div>
</ul>
`;
